doc/cluster_design.md

```diff
@@ -0,0 +1,61 @@
+# Design: Clustering in SQLflow to analyze patterns in data
+
+## Concept
+
+For analysts and real business people, in the daily analysis work, most of the work is not prediction, but analysis of the patterns in the data. This can help them mine user behavioral characteristics and differences, helping the business discover value and operate.
+
+This design doc introduces how to support the `Cluster Model` in SQLFlow. 
```

Comment:
Add a section to introduce the `Cluster Model`?

Reply:
> As one of the most powerful method  in  pattern recognition, clustering focus on finding the similarities between items per group and differences between groups. Hence, `Cluster Model` can help analysts to build such a model which can split data samples into different group according to their features automatically.   

@Echo9573 If it can help.